File: websocket网络相机/服务端代码/public/daikuan.js
function Daikuan(dom) {
    this.dom = dom;

    this.lastTime = 0;
    this.now = 0;
    this.sizeCount = 0;
    this.update = function (size) {
        this.now = performance.now(); //单位毫秒，比Date.now()更精确
        this.sizeCount += size; //记录1秒钟内，累计接收到的数据包大小
        if (this.now - this.lastTime >= 1000) {
            //如果已经过去1秒，则计算每秒能传输多少字节数据
            const speed = this.sizeCount / (this.now - this.lastTime) * 1000;
            dom.innerHTML = `
                <p>网速1：${Math.floor(speed)} Byte/s | ${Math.floor(speed / 1024)} KByte/s | ${(speed / 1024 / 1024).toFixed(2)} MByte/s</p>
                <p>网速2：${Math.floor(speed * 8)} bit/s | ${Math.floor(speed * 8 / 1024)} Kbit/s | ${(speed * 8 / 1024 / 1024).toFixed(2)} Mbit/s</p>
            `;
            this.sizeCount = 0;
            this.lastTime = this.now;
        }
    }
}

File: esp32性能测试/esp32代码/daikuan.py
class Daikuan:
    datetimeModule = None  # datetime模块
    osModule = None  # os模块
    printCount = 0  # 打印次数（测试次数）

    lastTime = 0
    now = 0
    sizeCount = 0

    def __init__(self):
        # self.datetimeModule = __import__("datetime")
        # self.osModule = __import__("os")
        self.datetimeModule = __import__("time")

    def get_timestamp(self):
        # return self.datetimeModule.datetime.now().timestamp() * 1000  # 将秒转成毫秒
        return self.datetimeModule.ticks_ms()

    def clear_print(self):
        # if self.osModule.name == 'nt':
        #     self.osModule.system("cls")  # 以下适用于Windows系统
        # else:
        #     self.osModule.system("clear")  # 以下适用于Linux/Mac系统
        return None
    
    def update(self, size):
        self.now = self.get_timestamp()

        self.sizeCount += size  # //记录1秒钟内，累计接收到的数据包大小

        if self.now - self.lastTime >= 1000:
            # 如果已经过去1秒，则计算每秒能传输多少字节数据
            speed = (self.sizeCount / (self.now - self.lastTime)) * 1000

            # self.clear_print()

            print("网速1："+str(round(speed, 2))+" Byte/s | "+str(round(speed / 1024, 2))+" KByte/s | "+str(round(speed / 1024 / 1024, 2))+" MByte/s")
            print("网速2："+str(round(speed * 8, 2))+" bit/s | "+str(round(speed * 8 / 1024, 2))+" Kbit/s | "+str(round(speed * 8 / 1024 / 1024, 2))+" Mbit/s")

            self.sizeCount = 0
            self.lastTime = self.now

            self.printCount += 1  # 记录打印次数

'''
daikuan = Daikuan()
isLoop = True
chunk = 'a'.encode()
while isLoop:
    daikuan.update(len(chunk))
    if daikuan.printCount >= 10:
        isLoop = False
'''